scipy/optimize/_highs/setup.py

```diff
@@ -138,13 +137,13 @@ def configuration(parent_package='', top_path=None):
     # Export constants and enums from HiGHS:
     ext = config.add_extension(
         '_highs_constants',
-        sources=['cython/src/_highs_constants.cxx'],
+        sources=[join('cython', 'src', '_highs_constants.cxx')],
         include_dirs=[
-            'cython/src/',
-            'src/',
-            'src/io/',
-            'src/lp_data/',
-            'src/simplex/',
```

Comment:
I'm surprised this actually worked. Using `/` with `pathlib` objects is fine, but usually not in hardcoded paths. We use `os.path.join` everywhere else and avoid platform-specific strings.

Reply:
AFAIK, `pathlib` will always do the right thing with path separators (that's what I always default to it), but yes, I agree better to be consistent with scipy convention

Reply:
Yes indeed, my point was that this is not using `pathlib`.